### ISSUE TITLE
NaN gradient issue when training with larger batch size and learning rate

### ISSUE THREAD

**Issue by roginsteam** [YvanYin/GoalFlow#40]
Thank you for your excellent work，
I attempted to run the training script on a single GPU. To reduce the total training time, I modified the hyperparameters as follows:
Batch Size: 16 (Original: 2)
Learning Rate: 2e-4 (Original: 1e-4)
However, I encountered an issue where the gradients (and subsequently the Flow Matching loss) suddenly became NaN during the later stages of training.
I would like to ask:
1. Is the model sensitive to the "Batch Size to Learning Rate" scaling ratio?
2. Are there any specific recommendations for numerical stability (e.g., specific AMP settings or gradient clipping thresholds) when training with these hyperparameters?
Thank you for your help!
<img width="1156" height="426" alt="Image" src="https://github.com/user-attachments/assets/317cef03-1d52-4f88-8283-c30dab17f5e8" />
